unit-testing/presentation.html

```diff
@@ -1,27 +1,310 @@
+        <pre><code data-trim data-noescape>
+            {​​
+              "scripts": {​​
+                "test": "jest",​​
+              },​​
+              "jest": {​​
+                "preset": "react-native",​​
+              }​​
+            }
+          </code></pre>
+      </section>
+      <!-- 15 slide -->
+      <section>
+        <h3 class="slide-name">jest.config.js and jest.setup.js files​</h3>
+        <p class="font-size-2 align-center">
+          A list of paths to modules that run some code to configure or set up the testing environment. Each setupFile
+          will be run once per test file.​​​</p>
+        <pre><code data-trim data-noescape>
+            module.exports = {​​
+              roots: ['<rootDir>/src'],​​
+              collectCoverage: true,​​
+              coverageThreshold: {​​ ... },​​
+              preset: 'react-native',​​
+              setupFiles: ['<rootDir>jest.setup.js'],​​
+              setupFilesAfterEnv: ['<rootDir>/src/globalMock.js'],​​
+            };
+          </code></pre>
+      </section>
+      <!-- 16 slide -->
+      <section>
+        <h3 class="slide-name">Structuring Tests​</h3>
+        <pre><code data-trim data-noescape>
+          beforeAll(() => {​
+            // ...​​
+          });​​
+          afterAll(() => {​​
+            // ...​​
+          });​​
+          describe('Our component', () => {​​
+            beforeEach(() => {​​
+             // ...​​
+            });​​
+            it('should test this', () => {​​
+             // ...​​
+            });​​
+          });
+          </code></pre>
+      </section>
+      <!-- 17 slide -->
+      <section>
+        <h3 class="slide-name">Expect​</h3>
+        <pre><code data-trim data-noescape>
+          expect(value)​​
+          expect.anything()​​
+          expect.any(constructor)​​
+          .not​​
+          .toBe(value)​​
+          .toHaveBeenCalled()​​
+          .toBeLessThan(number | bigint)​​
+          .toBeNull()​​
+          .toContain(item)​​
+          .toEqual(value)​​
+          .toMatch(regexp | string)​​
+          .toMatchSnapshot(propertyMatchers?, hint?)​​
+          .toThrow(error?)​​
+        </code></pre>
+      </section>
+      <!-- 18 slide -->
+      <section>
+        <h3 class="slide-name">Mocking & The Jest Object​</h3>
+        <pre><code data-trim data-noescape>
+          // Mock functions​​
+          jest.mock(moduleName, factory)​​
+          jest.unmock(moduleName)​​
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+      </section>
+      <!-- 19 slide -->
+      <section>
+        <h3 class="slide-name">Snapshots​​</h3>
+        <pre><code data-trim data-noescape>
```

Comment:
could you add proc / cons of this approach?

Reply:
done